<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialogForm</class>
 <widget class="QDialog" name="SettingsDialogForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>725</width>
    <height>421</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="4" column="1" colspan="2">
        <widget class="QLineEdit" name="downloadsLocation"/>
       </item>
       <item row="1" column="2">
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>280</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Home:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QLineEdit" name="home"/>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="goToHomePageOnOpen">
         <property name="text">
          <string>Go to home page when opening a new browser tab</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1" colspan="2">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>391</width>
           <height>262</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Save downloads to:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QPushButton" name="changeDownloadsPathButton">
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Change directory</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="title">
       <string>IPFS</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="sizeConstraint">
          <enum>QLayout::SetNoConstraint</enum>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBoxDaemon">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Spawn IPFS daemon and use as transport</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="2" column="0">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>HTTP gateway port</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="ipfsdApiPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="ipfsdGwPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Minimum swarm connections</string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QSpinBox" name="swarmMinConns">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>Maximum swarm connections</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="ipfsdSwarmPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>API port</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QSpinBox" name="swarmMaxConns">
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>300</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Swarm port</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Maximum storage (Gb)</string>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QSpinBox" name="storageMax">
              <property name="minimum">
               <number>5</number>
              </property>
              <property name="maximum">
               <number>2048</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxIpfsConn">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Use a custom IPFS daemon</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Host</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>API port</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="customIpfsApiPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>HTTP gateway port</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="customIpfsGwPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="customIpfsHost">
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="maximumSize">
            <size>
             <width>721</width>
             <height>145</height>
            </size>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Restart the application after changing the IPFS connection settings.</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>UI</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>File manager</string>
           </property>
           <widget class="QCheckBox" name="wrapFiles">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>30</y>
              <width>661</width>
              <height>21</height>
             </rect>
            </property>
            <property name="text">
             <string>Wrap files with a directory (preserves filenames)</string>
            </property>
           </widget>
           <widget class="QCheckBox" name="wrapDirectories">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>60</y>
              <width>661</width>
              <height>21</height>
             </rect>
            </property>
            <property name="text">
             <string>Wrap directories with a directory</string>
            </property>
           </widget>
           <widget class="QCheckBox" name="hideHashes">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>90</y>
              <width>661</width>
              <height>21</height>
             </rect>
            </property>
            <property name="text">
             <string>Hide IPFS hashes in tree views</string>
            </property>
           </widget>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QGroupBox" name="groupBox_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Language</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>Language</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="language"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialogForm</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialogForm</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
